spec/models/ccms/submission_spec.rb

```diff
@@ -0,0 +1,280 @@
+              expect(history.to_state).to eq 'failed'
+              expect(history.success).to be false
+              expect(history.details).to match(/RuntimeError/)
+              expect(history.details).to match(/oops when adding/)
+            end
+          end
+
+          context 'failed response from CCMS adding an applicant' do
+            let(:applicant_search_response) { File.read(File.join(Rails.root, 'spec', 'services', 'ccms', 'data', 'applicant_search_response_no_results.xml')) }
+            let(:applicant_add_response) { File.read(File.join(Rails.root, 'spec', 'services', 'ccms', 'data', 'applicant_add_response.xml')) }
+            let(:transaction_request_id_in_example_response) { '20190301030405123456' }
+            before do
+              applicant_search_requestor = double ApplicantSearchRequestor
+              allow(submission).to receive(:applicant_search_requestor).and_return(applicant_search_requestor)
+              expect(applicant_search_requestor).to receive(:call).and_return(applicant_search_response)
+              expect(applicant_search_requestor).to receive(:transaction_request_id).and_return(transaction_request_id_in_example_response)
+              applicant_add_requestor = double ApplicantAddRequestor
+              expect(ApplicantAddRequestor).to receive(:new).with(legal_aid_application.applicant).and_return(applicant_add_requestor)
+              expect(applicant_add_requestor).to receive(:call).and_return(applicant_add_response)
+              expect(applicant_add_requestor).to receive(:transaction_request_id).and_return(transaction_request_id_in_example_response)
+              applicant_add_response_parser = double ApplicantAddResponseParser
+              expect(ApplicantAddResponseParser).to receive(:new).and_return(applicant_add_response_parser)
+              expect(applicant_add_response_parser).to receive(:parse).and_return('Failure')
+            end
+
+            it 'puts it into failed state' do
+              submission.process!
+              expect(submission.aasm_state).to eq 'failed'
+            end
+
+            it 'records the error in the submission history' do
+              expect { submission.process! }.to change { SubmissionHistory.count }.by(1)
+              history = SubmissionHistory.find_by(submission_id: submission.id)
+              expect(history.from_state).to eq 'case_ref_obtained'
+              expect(history.to_state).to eq 'failed'
+              expect(history.success).to be false
+              expect(history.details).to eq(applicant_add_response)
+            end
+          end
+        end
+      end
+    end
+
+    # private methods tested here because they are mocked out above
+    #
+    describe '#reference_data_requestor' do
+      it 'only instantiates one copy of the ReferenceDataRequestor' do
+        sub = Submission.new
+        requestor1 = sub.__send__(:reference_data_requestor)
+        requestor2 = sub.__send__(:reference_data_requestor)
```

Comment:
Why is `__send__` being used with simple static content? 

Also within spec variables should be defined via `let`

I'd suggest:
```ruby
let(:requestor1) { sub.reference_data_requestor }
let(:requestor2) { sub.reference_data_requestor }
```

Reply:
`reference_data_requestor` is a private method. `sub.reference_data_requestor` fails with a scoping error.

(Whether it should be private in the first place is another question...)

I've moved the definitions to a `let` block though.